### PR TITLE
chore : refactor code quality issues

### PR DESCRIPTION
**Pull request by powerexploit**
## Description
Hey 👋 , I'm member of the Developer Outreach team at DeepSource and ran DeepSource analysis on my fork of the repo. It found some interesting [code quality improvements](https://deepsource.io/gh/ankitdobhal/analytics-zoo/issues) to consider.

This PR fixes a few of the issues detected for you to assess if it is right for you.
Happy to provide the tweaks separately otherwise :)

### What problem does this PR solve?

This PR improve some code quality issues like anti-pattern and bug risks. 

### The fix

- Merge `isinstance` calls
- Remove duplicate elements during `set` declaration
- Remove implicit `object` from the base class
- Remove unnecessary `typecast`
- Use `sys.exit()` calls
- Added `.deepsource.toml` file for continuous analysis on bug risks/performance/code-quality issues on new changes.

### CODE REVIEW

**Comment by qiyuangong** [comment]
Hi @ankitdobhal 
Thank you for contribution! Can you resolve conflicts with rebase? such that we can merge it into master after review.

**Comment by hkvision** [comment]
Quite good!

**Comment by yushan111** [comment]
@ankitdobhal 
Thanks a lot for your contribution!

**Review by yushan111 — APPROVED** [review]

File: .deepsource.toml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+version = 1
```
Comment (hkvision) [inline .deepsource.toml:1]: remove this file?

**Comment by ankitdobhal** [comment]
> Hi @ankitdobhal
> Thank you for contribution! Can you resolve conflicts with rebase? such that we can merge it into master after review.

Hey , @qiyuangong Thanks for response.
But I don't see any conflict issue as per github pointing  out .
Can you help me out in this ?

**Comment by yushan111** [comment]
@ankitdobhal 
hi, you may need to update upstream master before rebase. Below are some commands you can refer to.
```bash
git checkout master(upstream master) 
git pull
git checkout fix-bug
git rebase master
# you may need to solve the conflicts
git push -f
```

Thanks

**Comment by hkvision** [comment]
After including this file `.deepsource.toml`, is there any way/script for us to do the check periodically on our own environment?

**Pull request closed without merging**